File: typescript/base-hmi-types/src/screens/controls/HmiAlarmDisplayFilter.ts
import { HmiProperty } from "../base.js";
import { HmiAlarmFilterExpression } from "./HmiAlarmFilterExpression.js";

export class HmiAlarmDisplayFilter {
  name?: string;
  definition?: HmiProperty<string>;
  expression?: HmiAlarmFilterExpression;
  isInitial?: HmiProperty<boolean>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmDisplayFilter.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmDisplayFilter
{
    public string? Name { get; set; }

    public HmiProperty<string>? Definition { get; set; }

    public HmiAlarmFilterExpression? Expression { get; set; }

    public HmiProperty<bool>? IsInitial { get; set; }
}
